components/timezone/src/ids.rs

```diff
@@ -0,0 +1,677 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use alloc::borrow::Cow;
+use alloc::string::String;
+use alloc::vec::Vec;
+use icu_provider::prelude::*;
+use zerotrie::cursor::ZeroAsciiIgnoreCaseTrieCursor;
+
+use crate::{
+    provider::names::{
+        Bcp47ToIanaMapV1, Bcp47ToIanaMapV1Marker, IanaToBcp47MapV2, IanaToBcp47MapV2Marker,
+    },
+    TimeZoneBcp47Id, TimeZoneError,
+};
+
+/// A mapper between IANA time zone identifiers and BCP-47 time zone identifiers.
```

Comment:
For developers unfamiliar with the details of the IANA Time Zone Database (which is most of them!), it may be helpful to understand the scale of the timezone ID data, in order to help them understand the perf and RAM consequences of various mapping options. Here's one possible way to do it. Feel free to ignore if this kind of info is not appropriate here.

```suggestion
/// A mapper between IANA time zone identifiers and BCP-47 time zone identifiers.
///
/// There are about 600 IANA time zone identifiers, and fewer than 500 BCP-47
/// time zone identifiers.
///
/// BCP-47 time zone identifiers are 8 ASCII characters or less and currently 
/// average 5.1 characters long. Current IANA time zone identifiers are less than
/// 40 ASCII characters and average 14.2 characters long.
///
/// These lists grow very slowly; in a typical year, 2-3 new identifiers are added.
```